tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs

```diff
@@ -0,0 +1,157 @@
+            events.Add(event3);
+
+            // combine adjacent acoustic events
+            events = AcousticEvent.CombineOverlappingEvents(events, segmentStartOffset);
+
+            Assert.AreEqual(2, events.Count);
+            Assert.AreEqual(1.0, events[0].EventStartSeconds, 1E-4);
+            Assert.AreEqual(6.5, events[0].EventEndSeconds, 1E-4);
+            Assert.AreEqual(7.0, events[1].EventStartSeconds, 1E-4);
+            Assert.AreEqual(9.0, events[1].EventEndSeconds, 1E-4);
+
+            Assert.AreEqual(1000, events[0].LowFrequencyHertz);
+            Assert.AreEqual(8000, events[0].HighFrequencyHertz);
+            Assert.AreEqual(1000, events[1].LowFrequencyHertz);
+            Assert.AreEqual(8000, events[1].HighFrequencyHertz);
+
+            Assert.AreEqual(5.0, events[0].Score, 1E-4);
+            Assert.AreEqual(9.0, events[1].Score, 1E-4);
+            Assert.AreEqual(0.5, events[0].ScoreNormalised, 1E-4);
+            Assert.AreEqual(0.9, events[1].ScoreNormalised, 1E-4);
+        }
+
+        [TestMethod]
+        public void TestSonogramWithEventsOverlay()
+        {
+            // make a substitute sonogram image
+            int width = 100;
+            int height = 256;
+            var substituteSonogram = new Image<Rgb24>(width, height);
+
+            //substituteSonogram.Mutate(x => x.Pad(width, height, Color.Gray));
+            // image.Mutate(x => x.Resize(image.Width / 2, image.Height / 2));
+
+            // make a list of events
+            var framesPerSecond = 10.0;
+            var freqBinWidth = 43.0664;
+            var segmentStartOffset = TimeSpan.Zero;
+            var minHz = 1000;
+            var maxHz = 8000;
+            double maxPossibleScore = 10.0;
+
+            var events = new List<AcousticEvent>();
+            var startTime1 = 1.0;
+            var duration1 = 5.0;
+            var event1 = new AcousticEvent(segmentStartOffset, startTime1, duration1, minHz, maxHz)
+            {
+                Score = 10.0,
+                Name = "Event1",
+                ScoreNormalised = 10.0 / maxPossibleScore,
+            };
+
+            events.Add(event1);
+            var startTime2 = 7.0;
+            var duration2 = 2.0;
+            var event2 = new AcousticEvent(segmentStartOffset, startTime2, duration2, minHz, maxHz)
+            {
+                Score = 1.0,
+                Name = "Event2",
+                ScoreNormalised = 1.0 / maxPossibleScore,
+            };
+            events.Add(event2);
+
+            // now add events into the spectrogram image.
+            // set colour for the events
+            foreach (AcousticEvent ev in events)
+            {
+                ev.BorderColour = AcousticEvent.DefaultBorderColor;
+                ev.ScoreColour = AcousticEvent.DefaultScoreColor;
+                ev.DrawEvent(substituteSonogram, framesPerSecond, freqBinWidth, height);
+            }
+
+            //substituteSonogram.Save("C:\\temp\\image.png");
```

Comment:
This is a bad example to leave in. Remove or fix path. See `GeneratedImageTests` for example output save directory